pprl/layers/tests/test_param_name.py

```diff
@@ -0,0 +1,68 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import pprl.layers as layers
+
+
+class TestParamName(unittest.TestCase):
+    def test_name_number(self):
+        self.fc1 = layers.fc(100)
+        self.fc2 = layers.fc(100)
+        self.fc3 = layers.fc(100, bias_attr=False)
+        self.fc4 = layers.fc(100, param_attr=False)
+        self.fc5 = layers.fc(100, name="fc", bias_attr=False)
+        self.embedding = layers.embedding((100, 128))
+        self.embedding_custom = layers.embedding(
+            (100, 128), name="embedding_custom")
+        self.conv2d = layers.conv2d(
+            num_filters=64,
+            filter_size=3,
+            name="my_conv2d",
+            set_paras=self.embedding.parameters())
```

Comment:
This sharing is illegal. The size of embedding is 100*128, the conv2d parameter size is undefined at this moment.